### ISSUE TITLE
Markdown to HTML: GitHub blob image URL is not being converted properly

### ISSUE THREAD

**Issue by jschleus** [jgm/pandoc#6952]
If one converts the [README.md]( https://raw.githubusercontent.com/pytorch/pytorch/master/README.md) file of the project [pytorch](https://pytorch.org/) 1.7.1 (hosted on GitHub) from Markdown to HTML using pandoc version 2.11.2 with the option `--from=gfm` (GitHub-Flavored Markdown) the leading Markdown line containing the project logo
```
![PyTorch Logo](https://github.com/pytorch/pytorch/blob/master/docs/source/_static/img/pytorch-logo-dark.png)
```
is converted into
```
<p><img src="https://github.com/pytorch/pytorch/blob/master/docs/source/_static/img/pytorch-logo-dark.png" alt="PyTorch Logo" /></p>
```
But the original URL doesn't deliver a file in PNG format  but GitHub related HTML code so instead of the logo the alternative text "PyTorch Logo" is displayed by browsers.

First I thought that accidentally an erroneous URL was specified in the README.md file but viewing the [HTML converted README.md](https://github.com/pytorch/pytorch/blob/master/README.md)  on GitHub itself the logo appears correctly at the top and in the according source code one can find
```
<p><a target="_blank" rel="noopener noreferrer" href="https://github.com/pytorch/pytorch/blob/master/docs/source/_static/img/pytorch-logo-dark.png"><img src="https://github.com/pytorch/pytorch/raw/master/docs/source/_static/img/pytorch-logo-dark.png" alt="PyTorch Logo" style="max-width:100%;"></a></p>
```
So the string `/blob/` in the original specified URL path is replaced by `/raw/` what leads to an URL that redirects to the desired PNG logo. After doing that replacement subsequently in the `pandoc` HTML output the logo is as expected also displayed.

I'm not sure if that is a pandoc flaw or a very specific GitHub behavior that isn't supported by pandoc.

**Comment by jgm** [comment]
Looks like specific GitHub behavior; instead of generating a simple img as the spec requires, they are creating an image within a link; the link points to the URL given and the image uses a URL they must be generating on the fly.

I'm not sure if we should be trying to reproduce all of that.  But it looks as if the transformation of the URL is simple enough: substitute `raw` for `blob`.  We'd have to recognize URLs of this kind and do the change.  Maybe under an extension?

**Comment by ickc** [comment]
I'd regard this as an upstream problem (not GitHub but PyTorch). I think they write the URL incorrectly.

Using pandoc as an example:

If you click into the README, you enter this URL: <https://github.com/jgm/pandoc/blob/master/README.md>, which is showing you this file in the GitHub interface.

If you click the raw button on the top right of the main panel, you got to the original README.md: <https://github.com/jgm/pandoc/raw/master/README.md>.

Somehow PyTorch dev linked to the wrong link (<https://github.com/pytorch/pytorch/blob/master/docs/source/_static/img/pytorch-logo-dark.png> is not a png but html! Download it and you'll see `<!DOCTYPE html>...`), and the GitHub markdown engine automatically convert it for them.

Just like the issue tracker markdown engine, if you have a link `https://github.com/jgm/pandoc/issues/6952`, it will automatically changed it to https://github.com/jgm/pandoc/issues/6952 if you are in the same repo, and `jgm/pandoc#6952`. It is part of their user experience magic, but are surprise behavior.

So in short I think pandoc should not be coded to correct this kind of mistakes.

**Comment by albert-github** [comment]
Seen the problems that doxygen and pandoc have with this link I created an upstream issue: https://github.com/pytorch/pytorch/issues/49475
